### PR TITLE
docs: add developers guide

### PR DESCRIPTION
**Pull request by spirosdelviniotis**
* Adds documentations rules
* Adds frameworks overview
* Adds git
* Adds inspire record
* Adds invenio
* Adds pidstore
* Adds workflow

Signed-off-by: Spiros Delviniotis <spyridon.delviniotis@cern.ch>

### CODE REVIEW

**Comment by david-caro** [comment]
I want to do some modifications to the diagrams, can you add also the source code for them? (the xml files or something)
Also can you add a note in the docs that the diagrams where edited with draw.io? so someone else can modify them in the future. We should find also some way to have an offline copy of the app, so if the service disappears we can still edit them.

**Comment by david-caro** [comment]
The tool looks really nice btw :) (and gplv3!)

**Comment by david-caro** [comment]
I modified the structure and added completed some missing info, still incomplete though, but I don't want to lose the changes, and well, allow review and someone else picking it up if wanted.

File: docs/workflows.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+.. code-block:: shell
+
+    inspirehep shell
+
+
+.. code-block:: python
+
+    from invenio_workflows import workflow_object_class, ObjectStatus
+    errors = workflows_object_class.query(status=ObjectStatus.ERROR)
```
Comment (david-caro) [inline docs/workflows.rst]: s/workflows_object_class/workflow_object_class/

**Comment by jacquerie** [comment]
Same as https://github.com/inspirehep/inspire-next/pull/1944#issuecomment-309744476.

**Pull request merged**